core/space/services/services_fs.go

```diff
@@ -492,6 +493,22 @@ func (s *Space) addFile(ctx context.Context, sourcePath string, targetPath strin
 		return domain.AddItemResult{}, err
 	}
 
+	if s.tc.IsBucketBackup(ctx, b.Slug()) && !s.tc.IsMirrorFile(ctx, targetPath, b.Slug()) {
```

Comment:
This logic might be better off inside textile client rather than the service. My thinking is that this pretty much only accesses textile client functions (`UploadFileToHub`, `MarkMirrorFileBackup`), but these ops seem internal concerns of the textile client rather than stuff that should be called from outside.

Maybe we can expose a single `tc.AddFile` that internally checks for these flags (`IsBucketBackup`) and then chooses wether to mirror or not. That way all these mirror functions can remain private.